beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -1049,27 +1049,26 @@ impl<AppReqId: ReqId, TSpec: EthSpec> Network<AppReqId, TSpec> {
     /// in Connected, Dialing or Banned state.
     fn dial_cached_enrs_in_subnet(&mut self, subnet: Subnet) {
         let predicate = subnet_predicate::<TSpec>(vec![subnet], &self.log);
-        let peers_to_dial: Vec<PeerId> = self
+        let peers_to_dial: Vec<Enr> = self
             .discovery()
             .cached_enrs()
             .filter_map(|(peer_id, enr)| {
                 let peers = self.network_globals.peers.read();
                 if predicate(enr) && peers.should_dial(peer_id) {
-                    Some(*peer_id)
+                    Some(enr.clone())
                 } else {
                     None
                 }
             })
             .collect();
-        for peer_id in peers_to_dial {
-            debug!(self.log, "Dialing cached ENR peer"; "peer_id" => %peer_id);
-            // Remove the ENR from the cache to prevent continual re-dialing on disconnects
-
-            self.discovery_mut().remove_cached_enr(&peer_id);
-            // For any dial event, inform the peer manager
-            let enr = self.discovery_mut().enr_of_peer(&peer_id);
-            self.peer_manager_mut().dial_peer(&peer_id, enr);
-        }
+
+        // Remove the ENR from the cache to prevent continual re-dialing on disconnects
+        // TODO: add this step to the iteration above.
```

Comment:
I'll add this guy also

Reply:
Is a bit trickier because the iterator is on a reference of cached peers. 

The second iteration isn't that bad. Feel free to update if you find a nice way to group them and avoid the ENR clone